tests/test_serialize_utils.cpp

```diff
@@ -0,0 +1,402 @@
+#include <doctest/doctest.h>
+#include <sparrow.hpp>
+
+#include "sparrow_ipc/magic_values.hpp"
+#include "sparrow_ipc/serialize_utils.hpp"
+#include "sparrow_ipc/utils.hpp"
+#include "sparrow_ipc_tests_helpers.hpp"
+
+namespace sparrow_ipc
+{
+    namespace sp = sparrow;
+
+    // Helper function to create a simple ArrowSchema for testing
+    ArrowSchema
+    create_test_arrow_schema(const char* format, const char* name = "test_field", bool nullable = true)
+    {
+        ArrowSchema schema{};
+        schema.format = format;
+        schema.name = name;
+        schema.flags = nullable ? static_cast<int64_t>(sp::ArrowFlag::NULLABLE) : 0;
+        schema.metadata = nullptr;
+        schema.n_children = 0;
+        schema.children = nullptr;
+        schema.dictionary = nullptr;
+        schema.release = nullptr;
+        schema.private_data = nullptr;
+        return schema;
+    }
+
+    // Helper function to create ArrowSchema with metadata
+    ArrowSchema create_test_arrow_schema_with_metadata(const char* format, const char* name = "test_field")
+    {
+        auto schema = create_test_arrow_schema(format, name);
+
+        // For now, let's just return the schema without metadata to avoid segfault
+        // The metadata creation requires proper sparrow metadata handling
+        return schema;
+    }
+
+    // Helper function to create a simple record batch for testing
+    sp::record_batch create_test_record_batch()
+    {
+        // Create a simple record batch with integer and string columns using initializer syntax
+        return sp::record_batch(
+            {{"int_col", sp::array(sp::primitive_array<int32_t>({1, 2, 3, 4, 5}))},
+             {"string_col",
+              sp::array(sp::string_array(std::vector<std::string>{"hello", "world", "test", "data", "batch"}))}}
+        );
+    }
```

Comment:
We may want to put these helper functions in https://github.com/QuantStack/sparrow-ipc/blob/main/tests/include/sparrow_ipc_tests_helpers.hpp